test/model/recent_senders_test.dart

```diff
@@ -0,0 +1,201 @@
+import 'package:checks/checks.dart';
+import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/events.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/model/recent_senders.dart';
+import '../example_data.dart' as eg;
+import 'recent_senders_checks.dart';
+
+void checkMatchesMessages(List<Message> messages, RecentSenders model) {
+  final Map<int, Map<int, MessageIdTracker>> expectedStreamSenders = {};
+  final Map<int, Map<String, Map<int, MessageIdTracker>>> expectedTopicSenders = {};
+  for (final message in messages) {
+    if (message is! StreamMessage) continue;
+
+    final StreamMessage(:streamId, :topic, :senderId, id: int messageId) = message;
+
+    // Track message in stream.
+    var sendersMap = expectedStreamSenders[streamId] ??= {};
+    var idTracker = sendersMap[senderId] ??= MessageIdTracker();
+    idTracker.add(messageId);
+
+    // Track message in topic.
+    final topicsMap = expectedTopicSenders[streamId] ??= {};
+    sendersMap = topicsMap[topic] ??= {};
+    idTracker = sendersMap[senderId] ??= MessageIdTracker();
+    idTracker.add(messageId);
+  }
+  check(model)
+    ..streamSenders.deepEquals(expectedStreamSenders)
+    ..topicSenders.deepEquals(expectedTopicSenders);
+}
+
+void main() {
+  group('RecentSenders', () {
+    test('starts with empty stream and topic senders', () {
+      final model = RecentSenders();
+      checkMatchesMessages([], model);
+    });
+
+    test('just process a stream message', () {
+      final model = RecentSenders();
+      final dmMessage = eg.dmMessage(from: eg.selfUser, to: [eg.otherUser]);
+      final streamMessage = eg.streamMessage(id: 100, sender: eg.user(userId: 10),
+        stream: eg.stream(streamId: 1), topic: 'topic');
```

Comment:
```suggestion
      final streamMessage = eg.streamMessage();
```

None of these details of the stream message are relevant to what the test is saying, right? It just needs them to have *some* plausible values.

That's then the job of `eg` aka `example_data.dart`. That file supplies boring but plausible data, so that tests can stay focused on expressing only the details that are relevant to the rest of the test.

(In the case of `eg.dmMessage` the `from` and `to` are required only because it's harder to come up with any truly boring canonical values for them to take — most tests do care whether it's a 1:1 or group DM, or whether it's to or from the self-user. And it's plausible only if the self-user is among the recipients, and the self-user in a given test case might not be `eg.selfUser`.)